src/components/liveSession/components/liveSessionBtn/liveSessionBtn.component.js

```diff
@@ -39,11 +48,18 @@
                 }
 
                 function showSessionModal() {
-                    $mdDialog.show({
-                        template: '<live-session-subject-modal student="vm.student"></live-session-subject-modal>',
-                        scope: $scope,
-                        preserveScope: true,
-                        clickOutsideToClose: true
+                    getScheduledLesson().then(scheduledLesson => {
```

Comment:
It is not acceptable that after clicking the "Start Session" button and util we get the lesson results from DocDB - the user will wait without seeing any indication of the app doing something.

Open a popup saying "Starting Lesson...".
Replace this popup when the lesson results return.